File: lab06c.py
#!/usr/bin/env python3
"""
Programming Assignment 6: Strange function
"""


def f(x):
    """
    A strange function.
    """
    if x > 100:
        return x - 10
    else:
        return g(f(x + 11))


def g(x):
    """
    Another strange function.
    """
    if x > 100:
        return x - 10
    else:
        return f(g(x + 11))


def test():
    """
    Test cases.
    """
    for x in range(200):
        print('f(' + str(x) + ') =', f(x))

# test()


print('on what input would you like me to run the function')
arg = input()
print('f(' + arg + ') =', f(int(arg)))

File: lab06c.cpp
//* Translate the PASCAL program described in the file lab06c.md into C++

//* Remove unused variables in the file lab06c.cpp

#include <iostream>

using namespace std;

auto g(int x) -> int;

auto f(int x) -> int {
    if (x > 100)
        return x - 10;
    else
        return g(f(x + 11));
}

auto g(int x) -> int {
    if (x > 100)
        return x - 10;
    else
        return f(g(x + 11));
}

int main() {
    cout << "on what input would you like me to run the function\n";
    auto arg = 0;
    cin >> arg;
    cout << "f(" << arg << ") = " << f(arg) << "\n";
    return 0;
}